docs/directives/needtable.rst

```diff
@@ -9,14 +9,16 @@ needtable
 
 .. code-block:: rst
 
-   .. needtable::
+   .. needtable:: Example table
       :tags: main_example
       :style: table
 
-.. needtable::
+.. needtable:: Example table
    :tags: main_example
    :style: table
 
+The argument of the ``needtable`` will be used as caption for the table. If argument is used, then 
```

Comment:
The argument of  ``needtable`` will be used as caption for the table. If argument is used, then 
the caption and table content will be wrapped together into a figure.

Have added/removed from above
```
- the
+ a
```

Reply:
done.